DAQ_FW/lib/IADCSensor/IADCSensor.h

```diff
@@ -0,0 +1,31 @@
+#ifndef IADCSENSOR_LIB
+#define IADCSENSOR_LIB
+
+#include "Logger.h"
+#include "system_config.h"
+
+class IADCSensor
+{
+private:
+    // bit to voltage conversion factor
+    float mVOLTAGE_PER_BIT;
+
+    // SENSOR METADATA:
+    // Currently Generic Parameters, to be expanded as needed
+    // TODO: add more specific data, depends on the sensors
+    const char *mSensorName;
+    uint32_t mSensorID;
+    uint32_t mADC_ID;
+
+    void ADCcovert(uint16_t _ADC_GAIN_MODE);
+
+public:
+    IADCSensor(const char *_SensorName, uint16_t _SensorID, uint16_t _ADC_ID, uint16_t _ADC_GAIN_MODE);
+
+    // Read function - to be overwritten by the Driver class
+    virtual void Read(){};
```

Comment:
This should either be abstract or have a default implementation to catch the mistake of a child not implementing this function. The former gives us an error at compilation and the latter logs the mistake at runtime. 

Abstract: `virtual void Read() = 0;` this means no child can be defined _unless_ an implementation for the `Read()` function is provided. This enforces every child to implement it

Default: If the function is not overridden the default implementation is used that could return an error or log the error to the console.

I think making the function abstract is the right call here I think since **every** sensor needs a `Read()` function to

Reply:
Is there a way to implement the abstract version currently without getting a compiler error? My understanding is that it needs at least one derived class with the method definition to run, which we currently don't have. I can add the error definition for the default version for now, and make it abstract once the derived class is there.

Reply:
Did you check that it doesn't compile? I believe the compiler only requires that any children must implement the method, **not** that there necessarily needs to be a derived class. If I'm wrong though you can do that for sure

Reply:
![image](https://github.com/sfuphantom/DAQ/assets/84810562/5acfd257-51da-4755-88e9-8d483af8d12f)
It gives this error, setting `virtual void Read() = 0;` makes the class abstract, thus the object from the base class cannot be declared.

Reply:
Ohh I see what you mean. Let's just remove that line of code. You can add it as a comment in the header file of the class if you want

Reply:
The current setup is fine. Disregard!